sceptre/template_handlers/formatted_yaml.py

```diff
@@ -0,0 +1,68 @@
+"""
+This module contains a template handler that can load files from disk and
+format the content.
+"""
+import re
+import os
+from sceptre.template_handlers.file import File
+
+
+class FormattedYaml(File):
+    """
+    Template handler that can load files from disk.
+    Supports JSON, YAML, Jinja2 and Python.
+
+    This class extends the `File` class from the
+    `sceptre.template_handlers.file` module. It provides additional
+    functionality to format YAML content by removing trailing whitespaces
+    and blank lines, adding line breaks before lines that don't start with
+    spaces, tabs, or specific prefixes, and adding a newline at the end
+    of the file.
+    """
+
+    # Class attribute for line start exceptions
+    no_newline_prefixes = ["---", "AWSTemplateFormatVersion", "Description"]
+
+    def handle(self):
+        """
+        Handles the template by returning the template contents.
+
+        Returns:
+            str or bytes: The template contents.
+        """
+        contents = super().handle()
+        if isinstance(contents, str):
+            return self._format_yaml_content(contents)
+        return contents
+
+    def _format_yaml_content(self, input_string: str) -> str:
```

Comment:
hmm, what if someone wants their own custom format?

Reply:
Having had some bad experiences in the matter in the past, I would probably advise them not to waste time on it! However, the option would be there to just subclass File as we have done and write their own template handler. Or perhaps better to just use the standard File template handler if specific output formatting is desired.